pkg/eniconfig/eniconfig.go

```diff
@@ -104,21 +119,34 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 
 	case *corev1.Node:
 
-		log.Infof("Handle corev1.Node: %s, %v", o.GetName(), o.GetAnnotations())
+		log.Infof("Handle corev1.Node: %s, %v, %v", o.GetName(), o.GetAnnotations(), o.GetLabels())
+		// Get annotations if not found get labels if not found fallback use default
 		if h.controller.myNodeName == o.GetName() {
 			annotation := o.GetAnnotations()
 
-			val, ok := annotation[eniConfigAnnotationDef]
+			val, ok := annotation[h.controller.eniConfigAnnotationDef]
 			if ok {
 				h.controller.eniLock.Lock()
 				defer h.controller.eniLock.Unlock()
 				h.controller.myENI = val
 				log.Infof(" Setting myENI to: %s", val)
 			} else {
-				h.controller.eniLock.Lock()
-				defer h.controller.eniLock.Unlock()
-				h.controller.myENI = eniConfigDefault
-				log.Infof(" Setting myENI to: %s", eniConfigDefault)
+
+				label := o.GetLabels()
+
+				val, ok := label[h.controller.eniConfigLabelDef]
+				if ok {
+					h.controller.eniLock.Lock()
+					defer h.controller.eniLock.Unlock()
+					h.controller.myENI = val
+					log.Infof(" Setting myENI to: %s", val)
+				} else {
+
+					h.controller.eniLock.Lock()
+					defer h.controller.eniLock.Unlock()
+					h.controller.myENI = eniConfigDefault
+					log.Infof(" Setting myENI to: %s", eniConfigDefault)
+				}
```

Comment:
I would much prefer this to look like @nak3 suggested in [his comment on #260](https://github.com/aws/amazon-vpc-cni-k8s/pull/260/files#r243752421)

Reply:
Added changes for cleaner logic. I agree it looks much better the way @nak3 suggested.